pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -707,118 +707,145 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		}
 	}
 
+	return ssc.rollingUpdateStatefulsetPods(
+		set, &status, currentRevision, updateRevision, revisions, pods, replicas, minReadySeconds,
+	)
+}
+
+func (ssc *defaultStatefulSetControl) rollingUpdateStatefulsetPods(
+	set *appsv1beta1.StatefulSet,
+	status *appsv1beta1.StatefulSetStatus,
+	currentRevision *apps.ControllerRevision,
+	updateRevision *apps.ControllerRevision,
+	revisions []*apps.ControllerRevision,
+	pods []*v1.Pod,
+	replicas []*v1.Pod,
+	minReadySeconds int32,
+) (*appsv1beta1.StatefulSetStatus, error) {
+
 	// for the OnDelete strategy we short circuit. Pods will be updated when they are manually deleted.
 	if set.Spec.UpdateStrategy.Type == apps.OnDeleteStatefulSetStrategyType {
-		return &status, nil
```

Comment:
consider move this checking out of the function, so that the function only deal with the case of rolling update.

Reply:
moved in  [d4cecdd18060f7bcc3eeb97364ea3d77b93782d7..46e4e4111c32ab60a122f3e0fcdc5e9a74628bdd](https://github.com/openkruise/kruise/compare/d4cecdd18060f7bcc3eeb97364ea3d77b93782d7..46e4e4111c32ab60a122f3e0fcdc5e9a74628bdd)